content/en/blog/2022-02-28-storing-charts-in-oci.md

```diff
@@ -0,0 +1,92 @@
+
+```console
+$ helm create demo
+Creating demo
+
+$ helm package demo
+Successfully packaged chart and saved it to: /tmp/demo-0.1.0.tgz
+
+$ echo "mypass" | helm registry login r.example.com -u myuser --password-stdin
+Login Succeeded
+
+$ helm push demo-0.1.0.tgz oci://r.example.com/myuser
+Pushed: r.example.com/myuser/demo:0.1.0
+Digest: sha256:7ed393daf1ffc94803c08ffcbecb798fa58e786bebffbab02da5458f68d0ecb0
+```
+
+More detail on [working with registries can be found in the Helm documentation](https://helm.sh/docs/topics/registries/).
+
+## The Helm SDK
+
+The Helm SDK, which is useful for those building tools on top of Helm functionality, also includes that ability to work with registries programatically. The following example illustrates pushing a chart to a registry:
+
+```go
+package main
+
+import (
+    "helm.sh/helm/v3/pkg/registry"
+)
+
+func main() {
+    client := ...
+
+    ref := "r.example.com/myuser/demo:0.1.0"
+
+    info, err := client.Push(b, ref)
+    if err != nil {
+        panic(err)
+    }
+}
+```
+
+More detail can be found in the [documentation for the registry package](https://pkg.go.dev/helm.sh/helm/v3/pkg/registry).
+
+## Limitations
+
+There are some limitations when using registries to store charts compared to Helm repositories or storing container images in registries.
+
+Helm repositores can be added and searched from the local Helm client. This is similar to how repositories work with other package managers such as zypper or apt. When working with OCI registries, this is not an option. OCI based registries don't provide standard APIs to facilitate searching.
+
+While the OCI specification provides for artifacts, not all registries support storing Helm charts or other artifacts that are not container images. Before choosing a registry you need to make sure it supports storing Helm charts.
+
+## Artifact Hub Support
+
+[Artifact Hub](https://artifacthub.io/), another CNCF project, provides a means to search and discover cloud native assets including charts. Helm charts stored in OCI based registries can be listed on Artifact Hub, which already knows how to work with them. More details on working with Artifact Hub and Helm charts in container registries can be found in their [documentation](https://artifacthub.io/docs/topics/repositories/#helm-charts-repositories).
+
+## ORAS
+
+The [OCI Registry as Storage (ORAS) project](https://oras.land/), another CNCF project, provides the library Helm uses to work with registries. The ORAS project bills itself as:
+
+> Registries are evolving as generic artifact stores. To enable this goal, the ORAS project provides a way to push and pull OCI Artifacts to and from OCI Registries.
+
+If you want to work with other artifacts in registries the ORAS project may provide some tools to help you.
```

Comment:
```suggestion
If you want to work with other artifacts stored in registries the ORAS project may provide some tools to help you.
```

Reply:
The existing wording covers both storing artifacts in, and retrieving artifacts from registries, right?

To me this change might imply that ORAS only works with other artifacts that are already stored in registries 🤔